docs/proposals/dockerfile_component_detection.md

```diff
@@ -0,0 +1,58 @@
+# Enhance port detection for dockerfiles that are not in a component.
+
+## Related GitHub Issues
+* https://github.com/redhat-developer/alizer/issues/235
+
+## Background
+
+What is a component?
+
+Components are defined in the `languages-customization.yml` as a language that has specfic `configuration_files`.
+
+Example of a Go component:
+
+```yaml
+Go:
+  exclude_folders:
+    - "vendor"
+  configuration_files:
+    - "go.mod"
+  component: true
+```
+
+What happens when the repo being scanned has component(s) found but also contains a dockerfile one level outside of the repo?
+
+```shell
+.repo-being-detected-by-alizer
+├── component/
+│   └── dockerfile
+└── dockerfile
+```
+
+The dockerfile in the root level would not be used during analysis because it does not have a component definition in the `languages-customization.yml`
+
+## Potential Solution
+
+Introduce a new component field called `orphan_component` and add it to the `Dockerfile` language:
+
+```yaml
+Dockerfile:
+  configuration_files:
+    - "[Dd]ockerfile"
```

Comment:
We should also support Containerfile as well.

Reply:
Agreed!